internal/util/version/version.go

```diff
@@ -64,9 +69,12 @@ func init() {
 	info.BuildEnvironment = types.MustNewDocument()
 	for _, s := range buildInfo.Settings {
 		info.BuildEnvironment.Set(s.Key, s.Value)
+
 		switch s.Key {
 		case "vcs.revision":
-			info.Commit = s.Value
+			if s.Value != info.Commit {
```

Comment:
I like the way that this ensures that the version info which is used for builds is consistent with git.